lib/p2p/Pool.ts

```diff
@@ -633,6 +637,18 @@ class Pool extends EventEmitter {
       await this.handlePacket(peer, packet);
     });
 
+    peer.on('handshake', (oldHandshakeState) => {
+      // drop all orders for trading pairs that are no longer supported
+      if (oldHandshakeState) {
+        oldHandshakeState.pairs.forEach((pairId) => {
+          if (!peer.pairs!.includes(pairId)) {
+            // a trading pair was in the old handshake state but not in the updated one
```

Comment:
This doesn't need anything from `Pool`, so we can handle it entirely in `Peer`.

Reply:
I'm emitting `peer.dropPair` from `Pool` because other components are listening to `Pool` and not `Peer`. But if you're suggesting to move the logic to check which pairs have been dropped or added into `Peer` - in that case I suppose I'd have it emit specific events like `pairAdded` and `pairRemoved` rather than `handshake`?

Reply:
Yes, and then propagate it from `Pool`.